### PR TITLE
feat(reactor): migrate logic for changes to reactor

### PR DESCRIPTION
**Pull request by pollend**
PR for https://github.com/MovingBlocks/Terasology/issues/4798

Depends on:
- [ ] https://github.com/Terasology/Pathfinding/pull/62

### CODE REVIEW

**Comment by skaldarnar** [comment]
@pollend the title mentions RxJava - is this PR still up-to-date and/or relevant?

**Comment by pollend** [comment]
@skaldarnar I'll wait on the reactor changes

File: src/main/java/org/terasology/minion/move/FindPathToNode.java [hunk 1]
```diff
@@ -61,26 +65,19 @@ public void construct(final Actor actor) {
...
-                    moveComponent.path = paths.get(0);
-                }
-                actor.save(moveComponent);
-            }
-
-            @Override
-            public void onFailure(Throwable t) {
-                moveComponent.path = Path.INVALID;
-            }
-        }, MoreExecutors.directExecutor());
```
Comment (keturn) [inline src/main/java/org/terasology/minion/move/FindPathToNode.java:83]: the current version of FindPathToNode no longer has a usage of an Executor like this.

File: src/main/java/org/terasology/minion/work/WorkBoard.java [hunk 2]
```diff
@@ -22,14 +23,13 @@
 import org.terasology.engine.logic.selection.ApplyBlockSelectionEvent;
 import org.terasology.engine.registry.In;
 import org.terasology.engine.registry.Share;
-import org.terasology.engine.utilities.concurrency.Task;
-import org.terasology.engine.utilities.concurrency.TaskMaster;
```
Comment (keturn) [inline src/main/java/org/terasology/minion/work/WorkBoard.java:26]: The minion.work package has been removed (by #88).

**Review by keturn — COMMENTED** [review]
looks like this has now been replaced by #88

**Pull request closed without merging**